### PR TITLE
dvc: remote: optimize imports

### PR DESCRIPTION
**Pull request by efiop**
Decreases `time dvc --version` from ~1.4 sec to ~0.3 sec.

Related #2495
Related #2445

Signed-off-by: Ruslan Kuprieiev <ruslan@iterative.ai>

* [x] Have you followed the guidelines in our
      [Contributing document](https://dvc.org/doc/user-guide/contributing)?

* [x] Does your PR affect documented changes or does it add new functionality
      that should be documented? If yes, have you created a PR for
      [dvc.org](https://github.com/iterative/dvc.org) documenting it or at
      least opened an issue for it? If so, please add a link to it.

-----

### CODE REVIEW

**Comment by Suor** [comment]
The provider thing I very much don't like. It adds a layer of indirection, which makes code not locally understandable. I would prefer explicit dynamic imports.

**Comment by Suor** [comment]
Also, you import `paramiko` in `remote/ssh/connection.py`

**Comment by efiop** [comment]
@Suor Just to be clear, this PR was just an example to our discussion.

> Also, you import paramiko in remote/ssh/connection.py

But it is only dynamically imported in get_connection in RemoteSSH.

**Comment by efiop** [comment]
> The provider thing I very much don't like. It adds a layer of indirection, which makes code not locally understandable. I would prefer explicit dynamic imports.

That is fair. Adjusted. Please take a look.

File: dvc/remote/base.py [hunk 1]
```diff
@@ -86,31 +86,8 @@ class RemoteBASE(object):
...
-                "dependencies included: \n"
-                "        pip install 'dvc[all]'\n"
-                "\n"
-                "If you have installed dvc from a binary package and you "
-                "are still seeing this message, please report it to us "
-                "using https://github.com/iterative/dvc/issues. Thank you!"
-            ).format(url, missing, " ".join(missing), self.scheme)
-            raise RemoteMissingDepsError(msg)
+
+        self._check_deps(config)
```
Comment (Suor) [inline dvc/remote/base.py]:
```suggestion
        self._check_requires(config)
```

Comment (Suor) [reply dvc/remote/base.py]: Or rename `REQUIRES` to `DEPS`.

**Comment by Suor** [comment]
Another thing I would delay `_check_deps()` to the moment the thing is used.

File: dvc/remote/base.py (hunk 1 above)
Comment (efiop) [reply dvc/remote/base.py]: renamed to `_check_requires` for now, as it affects less code.

**Comment by efiop** [comment]
> Another thing I would delay _check_deps() to the moment the thing is used.

I don't think that is possible. For example RemoteSSH.__init__ will use _load_user_ssh_config, which needs to import paramiko, so dep check should stay where it is.

**Comment by efiop** [comment]
> BTW, we duplicated this effort)

Unless I've forgotten something, initially we've had a discussion about approaches to solve this and decided to do the research for this ticket first (this is what has been planned), then choose the approach and then do the PR. You've jumped to PR right away (which is totally understandable), and during its discussion, we've had a miscommunication, where I felt like you are overcomplicating the solution, so I had to do my own research and give a shot implementing this quick fix to illustrate my point.

**Pull request merged**